Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/overlay/InputOverlayPointer.java

```diff
@@ -17,13 +17,23 @@
   public static final int DOUBLE_TAP_2 = 2;
   public static final int DOUBLE_TAP_CLASSIC_A = 3;
 
+  public static final int MODE_DISABLED = 0;
+  public static final int MODE_FOLLOW = 1;
+  public static final int MODE_DRAG = 2;
+
   private final float[] axes = {0f, 0f};
+  private final float[] oldaxes = {0f, 0f};
 
   private float mGameCenterX;
   private float mGameCenterY;
   private float mGameWidthHalfInv;
   private float mGameHeightHalfInv;
 
+  private float mCenterX;
+  private float mCenterY;
```

Comment:
The naming of this pair of variables is confusing, for two reasons:

1. The reader might assume that since mGameCenter is the center of where the game surface is, mCenter is the center of the whole window. But it isn't.
2. While it does make sense to think of this as the "center" in a way when using drag mode, this variable also gets updated when in follow mode, and for follow mode it doesn't make sense to think of it that way.

Instead of `mCenter`, I would suggest something like `mDragStart` or `mTouchStart`.